### ISSUE TITLE
Could not find a Cypress configuration file. We looked but did not find a cypress.json file in this folder.

### ISSUE THREAD

**Issue by kahboom** [cypress-io/github-action#564]
As of late last week, Cypress tests are failing with this error, from one commit to the next, which was completely unrelated (an `.md` file). I seem to be getting the same error as @jazanne from [this comment](https://github.com/cypress-io/github-action/issues/430#issuecomment-1144054512).

We're using:
- `cypress-io/github-action@v2`
- Cypress v9.7.0
- Yarn Berry v3

I tried @jazanne's workaround, sadly we're still getting the same error regardless. We've tried upgrading cypress, using `cypress-io/github-action@v3`, and running it headlessly. In one separate PR I even tried to downgrade to yarn 1.x, which is causing some other errors. The whole JS toolchain seems so brittle. :s Is anyone else getting this? Any suggestions?

```
[STARTED] Task without title.
[SUCCESS] Task without title.
[494:0606/101147.234717:ERROR:node_bindings.cc(276)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
[494:0606/101147.234801:ERROR:node_bindings.cc(276)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
[494:0606/101147.234809:ERROR:node_bindings.cc(276)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
[494:0606/101147.234815:ERROR:node_bindings.cc(276)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[655:0606/101148.920788:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[655:0606/101148.926474:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Could not find a Cypress configuration file.
We looked but did not find a cypress.json file in this folder: /__w/kaoto-ui/kaoto-ui
Error: The process '/usr/local/bin/npx' failed with exit code 1
```

cc @delawen

- https://github.com/KaotoIO/kaoto-ui/runs/6753628560?check_suite_focus=true
- [GitHub actions workflow](https://github.com/KaotoIO/kaoto-ui/blob/main/.github/workflows/e2e-tests.yml)

**Comment by Groundhug** [comment]
I have a similar problem. We are using separate configuration files depending on whether the test should run on staging or in production, and then through the `package.json` file using the configuration file as a parameter when calling the appropriate command.

But I can only run the tests that do not include a configuration file as a parameter in the `package.json` file. All the tests with custom configuration files cannot run, and I get the same error.

```
Could not find a Cypress configuration file.
We looked but did not find a cypressStaging.config.json file in this folder: C:\_Stuff\Code\e2e
```

The files are present in the folder, with the name indicated in the error above. I don't know how to get Cypress to detect my configuration file.

I receive this error both through running the GitHub Action as well as when manually running the test using `npm run`.

**Comment by JuliusMu** [comment]
We got the same issue in a gitlab-ci pipeline. Looks like a driver issue to me: 
`libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)`
Any updates or workarounds yet?

**Comment by jdferreira2429** [comment]
same issue. anyone looking into this?

**Comment by SparrowBrain** [comment]
Same issue.

Using:
* cypress-io/github-action@v4
* Cypress v10.2.0
* npm (v8? GitHub Actions default)

Could be related to migrating to v2 of npm lockfile. Had to upgrade chokidar after migration.

The run:
https://github.com/SparrowBrain/GOG-Galaxy-Suggester/runs/7009698620

```
Run cypress-io/github-action@v2
/usr/local/bin/npm ci
added 2420 packages, and audited 2421 packages in 55s
15 packages are looking for funding
  run `npm fund` for details
28 vulnerabilities (1 moderate, 19 high, 8 critical)
To address all issues (including breaking changes), run:
  npm audit fix --force
Run `npm audit` for details.
/usr/local/bin/npx cypress cache list
┌─────────┬───────────┐
│ version │ last used │
├─────────┼───────────┤
│ 10.2.0  │ a day ago │
└─────────┴───────────┘
/usr/local/bin/npx cypress verify
[STARTED] Task without title.
[SUCCESS] Task without title.
/usr/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/GOG-Galaxy-Suggester/GOG-Galaxy-Suggester --files-from manifest.txt
Cache Size: ~55 MB (57565882 B)
Cache saved successfully
/usr/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/GOG-Galaxy-Suggester/GOG-Galaxy-Suggester --files-from manifest.txt
Cache Size: ~155 MB (162987616 B)
Cache saved successfully
start server "npm start command "npm start"
current working directory "/home/runner/work/GOG-Galaxy-Suggester/GOG-Galaxy-Suggester"
waiting on "http://localhost:3000" with timeout of 60 seconds
/usr/local/bin/npm start
> gog-galaxy-suggester@0.1.0 start
> react-app-rewired start
ℹ｢wds｣: Project is running at http://10.1.0.152/
ℹ｢wds｣: webpack output is served from /GOG-Galaxy-Suggester
ℹ｢wds｣: Content not from webpack is served from /home/runner/work/GOG-Galaxy-Suggester/GOG-Galaxy-Suggester/public
ℹ｢wds｣: 404s will fallback to /GOG-Galaxy-Suggester/
Starting the development server...
Compiled successfully!
You can now view gog-galaxy-suggester in the browser.
  Local:            http://localhost:3000/GOG-Galaxy-Suggester
  On Your Network:  http://10.1.0.152:3000/GOG-Galaxy-Suggester
Note that the development build is not optimized.
To create a production build, use npm run build.
[2203:0622/174911.935964:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[2203:0622/174911.939833:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Could not find a Cypress configuration file.
We looked but did not find a cypress.json file in this folder: /home/runner/work/GOG-Galaxy-Suggester/GOG-Galaxy-Suggester
Test run failed, code 1
More information might be available above
Error: Could not find Cypress test run results
Cypress module has returned the following error message:
Could not find Cypress test run results
```

The file has to be there, since github checkout points to the same folder

**Comment by drecali** [comment]
@SparrowBrain Thanks for sharing a link to your workflow logs. I had the same issue. In both our cases, you can see in the workflow logs that the `config-file` was set as `cypress.json` by default. According to the Cypress [docs](https://docs.cypress.io/guides/references/configuration#Common-problems:~:text=Reworked%20page%20to%20support%20new%20cypress.config.js%20and%20deprecated%20cypress.json%20files)

Version | Changes
-- | --
10.0.0 | Reworked page to support new `cypress.config.js` and deprecated `cypress.json` files

![image](https://user-images.githubusercontent.com/24983797/175286957-e9799d21-8f2f-4cd7-9f8b-7f2c815128f7.png)

I just changed the workflow to explicitly declare `config-file: cypress.config.ts` and it worked. In my case the CLI output was very helpful.

```bash
[percy] Running "npx cypress run --headless --config-file cypress.json --browser chrome"

You are attempting to use Cypress with an older config file: cypress.json

When you upgraded to Cypress v10.0 the config file was updated and moved to a new location: cypress.config.ts

You may need to update any CLI scripts to ensure that they are referring the new version. This would typically look something like:

"cypress open --config-file=cypress.config.ts"

https://on.cypress.io/migration-guide
```

**Comment by AshMcConnell** [comment]
> We got the same issue in a gitlab-ci pipeline. Looks like a driver issue to me: `libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)` Any updates or workarounds yet?

I have a similar issue, but it doesn't say it can't find the config file, instead  : -

```
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[491:0626/135959.592121:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[491:0626/135959.594708:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Can't run because no spec files were found.
We searched for specs matching this glob pattern:
```

**Comment by new-guy** [comment]
Yeah, I'm also getting this when I run it on my local docker.  `cypress info` ran in the cypress/included:9.7.0 container throws this error: ```root@23dfb8d855b9:/# cypress info
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[688:0628/153217.781946:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[688:0628/153217.788351:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Displaying Cypress info...```

**Comment by stychu** [comment]
Any solutions yet?? Seems like this is a wide issue as of now with cypress v10+.

**Comment by drecali** [comment]
@AshMcConnell @new-guy 

The errors below may not be related to this issue because they also appear in successful runs. Hope you're able to find the solution soon! 

```sh
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[491:0626/135959.592121:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[491:0626/135959.594708:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
```

**Comment by mikeerickson** [comment]
Ya, this sure seems like a widespread issue.  I have had to revert to using Cypress 9.5.4 as all the suggestions here have not worked for me.  If anybody knows of a working solution, please report here as we really want to Cypress 10.x

**Comment by snake-py** [comment]
+1 - inside docker:

```
[2122:0801/060102.754724:ERROR:bus.cc(397)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
[2122:0801/060102.754863:ERROR:bus.cc(397)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
[2122:0801/060102.756263:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2122:0801/060102.756321:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2276:0801/060102.822555:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[2276:0801/060102.825930:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
[2122:0801/060102.968302:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2122:0801/060102.988646:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
^C[2276:0801/060244.882776:ERROR:connection.cc(66)] X connection error received.
[2276:0801/060244.883598:ERROR:connection.cc(66)] X connection error received.
```

**Comment by gsouf** [comment]
Happens to me when upgrading from cypress 9.4 to cypress 10.4

**Comment by rozzs74** [comment]
Any workarounds?

**Comment by gsouf** [comment]
Actually this is working after running the cypress migration. The config file is not a json anymore. It is a JS file now. All I did is running `cypress open` to get cypress ui, and followed the migration from the UI. They migrate the cypress file. Then remember to  use `cypress run` with updated flag `--config-file` to point to the new js file

**Comment by javierfuentesm** [comment]
any ideas ? im getting this error when i want to use it with docker 
<img width="984" alt="image" src="https://user-images.githubusercontent.com/19244527/183240544-620d4488-75c8-4d8f-90c1-89524e4fc451.png">
but im using the 10.4.0 cypress version and i have the following configuration :

<img width="541" alt="image" src="https://user-images.githubusercontent.com/19244527/183240579-5c43dda2-0393-4918-ad56-411f42f9a8b6.png">

**Comment by javierfuentesm** [comment]
@gsouf any ideas ? with my problem, it seems like you solved it Im using the same version 
<img width="253" alt="image" src="https://user-images.githubusercontent.com/19244527/183240625-c558ea80-e502-4014-bd6b-0781b8f15a7a.png">
but getting the same error

**Comment by gsouf** [comment]
@javierfuentesm you issue seems to be different to this ticket. I think you need to put the full path from your app root (from where you invoke in the cypress command in `supportFile`. For example 

```
supportFile: 'cypress/support/e2e.js'
```

Or just remove this config because by default cypress will already pick the file at this location (https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file)

**Comment by javierfuentesm** [comment]
@gsouf  yes i tried that but is nto working , locally it works fine but on docker i get all those errors

**Comment by gsouf** [comment]
@javierfuentesm I got the same error as you did when I tried just `support/e2e.js` and it got resolved on docker when I use the config `cypress/support/e2e.js`.

That is weird, but overall I noticed many issues with path resolution when I tried to configure the new components testing today.

**Comment by arnabmondal19** [comment]
Is there any updates on this issue? i am also getting error while running in docker.
Cypress version: 10.3.0

![image](https://user-images.githubusercontent.com/54146331/184183260-9385e54a-f80d-4cc3-939d-1d478074f478.png)

**Comment by iliogr** [comment]
> Is there any updates on this issue? i am also getting error while running in docker. Cypress version: 10.3.0
> 
> ![image](https://user-images.githubusercontent.com/54146331/184183260-9385e54a-f80d-4cc3-939d-1d478074f478.png)

Having the same issue. Any solutions yet?

**Comment by gsouf** [comment]
@iliogr @arnabmondal19  this is not the same error. The error described in this ticket is `Could not find a Cypress configuration file. We looked but did not find a cypress.json file in this folder.`.

**Comment by iliogr** [comment]
> 

@gsouf I know. the error described in the ticket is caused because cypress v10 requires a different config file not cypress.json anymore.

however, I am interested in the issue mentioned above regarding dri3 extension not supported :)

**Comment by gsouf** [comment]
@iliogr I don't think that the "dri3 extension" message is an issue for me it's working well in docker and I'm getting the same error message about "dri extension".

Is your config file at the root of the your project? If not try to place it at the root of the project to see if it works. There are some ongoing issues with path resolutions when cypress config is not at the root. 

Also mentioning just in case you missed that.. with cypress 10 default specs files pattern is now using extension `*.cy.js`. Are your files matching this pattern?

Ultimately you can try to set the specPattern manually to an absolute path in your cypress config file: 

```js
const path = require('path');
const { defineConfig } = require('cypress');


module.exports = defineConfig({

  // for e2e testing
  e2e: {
    specPattern: path.resolve(__dirname, 'path/to/src/**/*.cy.js'), 
  },

  // for component testing
  component: {
    specPattern: path.resolve(__dirname, 'path/to/src/**/*.cy.js'), 
  },

})
```

**Comment by kodriscollics** [comment]
Having the same issue after upgrading to Cypress version 10. I am running inside of IntelliJ. When I click on a Cypress run configuration for a test, I get this message:

> Could not find a Cypress configuration file.
> 
> We looked but did not find a cypress.config.ts file in this folder: C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.2\jbr\bin

I have implemented the suggestions noted above... i.e. running with --config-file=cypress.config.ts in my test parameters, specPattern: 'cypress/e2e/**/*.{js,jsx,ts,tsx}' in the cypress.config.ts file, etc. The problem persists.

**Comment by gsouf** [comment]
Would you be able to provide a repo with a dockerfile that we can use to reproduce the issue @kodriscollics @iliogr ?

**Comment by kodriscollics** [comment]
I figured out the issue. After the upgrade, the value in all of the "Cypress project base:" fields for my Intellij tests were deleted. After I put the proper value back in the fields, the tests run without the error. Here is a look at what I'm talking about:

<img width="917" alt="CypressProjectBaseField" src="https://user-images.githubusercontent.com/75755773/185200174-89be5ddd-ca6c-43c8-a3c8-f682bc42fe0b.PNG">

**Comment by lucasfraga-bnex** [comment]
any news?

**Comment by JonnyWideFoot** [comment]
See: https://docs.cypress.io/guides/references/migration-guide#Cypress-Changes

"You may use the --config-file command line flag or the configFile module API option to specify a .js or .ts file. JSON config files are no longer supported."

It's working for me after changing:

/integration/config.json
```
{
  "baseUrl": "http://localhost:3000",
  "ignoreTestFiles": "**/config.json",
  "video": false,
  "experimentalSessionSupport": true
}

```

To:

/integration/config.js
```
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    baseUrl: 'http://localhost:3000',
    video: false,
    experimentalSessionSupport: true
  }
})

```

**Comment by rezasadeghi1** [comment]
God bless you @JonnyWideFoot !! Solved it all. Works for me now when running tests through kubernetes with a Helm Chart.

**Comment by w4dd325** [comment]
@drecali kudos, your comment "I just changed the workflow to explicitly declare config-file: cypress.config.ts and it worked." resolved my issue! Thank you.

**Comment by lalilaloe** [comment]
In my case I found that i had to use `cypress-io/github-action@v4` after getting the following error with v2

![image](https://user-images.githubusercontent.com/9057622/199705657-289f59c4-e8d4-4cd7-a2dd-c8ed2c7228eb.png)

**Comment by mosofsky** [comment]
In my case, I think the problem was I forgot to commit the new files Cypress 10 migration generated.  So Cypress ran successfully on my local computer but not on CI.

**Comment by Ayko1595** [comment]
Hi! Has anyone found a solution to this? We are running cypress in a docker container. Is it necessary to explicitly select the relevant config file?

**Comment by matart15** [comment]
Look like this problem still exist on `cypress-io/github-action@v5.0.2`

**Comment by MikeMcC399** [comment]
@matart15 

> Look like this problem still exist on cypress-io/github-action@v5.0.2

There are several different problems mixed up in this issue. Which error message are you getting and which version of Cypress are you using?

**Comment by matart15** [comment]
# version 

```
yarn cypress -v                                                                                                          
yarn run v1.22.19
warning package.json: No license field
$ /Users/matar/workplace/github/aisaac-lab/ticketx/node_modules/.bin/cypress -v
Cypress package version: 11.2.0
Cypress binary version: 11.2.0
Electron version: 21.0.0
Bundled Node version: 16.16.0
✨  Done in 0.66s.
```

# config 

```
// cypress.config.ts
import { defineConfig } from 'cypress'

export default defineConfig({
  e2e: {
    projectId: 'darxah',
    viewportWidth: 1920,
    viewportHeight: 1080,
    video: true,
    baseUrl: 'http://localhost:3000',
    chromeWebSecurity: false,
    watchForFileChanges: false,
    defaultCommandTimeout: 30000,
    setupNodeEvents(on, config) {
      require('@cypress/code-coverage/task')(on, config)
      on('before:browser:launch', (browser: any = {}, launchOptions) => {
        // `args` is an array of all the arguments that will
        // be passed to browsers when it launches
        console.log(launchOptions.args) // print all current args

        if (browser.name === 'chrome') {
          launchOptions.args.push('--disable-dev-shm-usage')
        }

        // whatever you return here becomes the launchOptions
        return launchOptions
      })
      return config
    },
  },
})

```


# problem

all tests Pass on my OSX.

But fails on CI. with this error 

```
[655:0606/101148.926474:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
```

# tried and same result

- `--broswer electron`
- `--browser chrome`
- `--disable-gpu`

**Comment by MikeMcC399** [comment]
@matart15 

> [4351:1227/131437.366155:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.

I believe that you can ignore the gpu error. I have seen successful tests which also show this error, for example run [6438334587](https://github.com/cypress-io/github-action/actions/runs/3786961899/jobs/6438334587) from this repository's examples.

Are you also getting a message like:

"Could not find a Cypress configuration file in this folder: /home/runner/work/xxxx"

Which runner are you using?

**Comment by MikeMcC399** [comment]
@matart15 

- I submitted a document change PR https://github.com/cypress-io/github-action/pull/689 so that the use of the `working-directory` parameter is better documented for v10 and later.
- You should check if you really need it in your configuration.
- It is used in all the examples in this repository and for the examples it works correctly.

**Comment by matart15** [comment]
@MikeMcC399  Thank you. 

Yes. Indeed I am using monorepo. I will try later and let you know the result

**Comment by MikeMcC399** [comment]
@matart15 

- I'm following up about the message `ERROR:gpu_memory_buffer_support_x11.cc` in https://github.com/cypress-io/cypress/issues/23326#issuecomment-1369831323

**Comment by matart15** [comment]
@MikeMcC399 
It looks like my team already using `working-directory`.
![Screenshot 2023-01-04 at 10 45 31](https://user-images.githubusercontent.com/17027219/210469561-6b6a1590-7dec-4c2f-a31d-88248ecef441.png)

this is screenshot of github action fail

![Screenshot 2023-01-04 at 10 49 36](https://user-images.githubusercontent.com/17027219/210469870-c018dce2-215d-4ccb-aaca-1886cb866801.png)

> ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
no other information

on my macbook tests succeed

**Comment by MikeMcC399** [comment]
@matart15 

Thank you for your screen shots and information. You mention only "ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported." This error occurs with Cypress even without using [cypress-io/github-action](https://github.com/cypress-io/github-action/). It does not look like it is preventing your test running on GitHub.

If this is the only error you are seeing, then I think you can ignore it. If you want to discuss it more in relation to [cypress-io/github-action](https://github.com/cypress-io/github-action/) please open a new issue. This issue is really about the error "Could not find a Cypress configuration file. We looked but did not find a cypress.json file in this folder."

**Comment by matart15** [comment]
@MikeMcC399 

I see. Thank you for explanation.

I will try running my test on CI without [cypress-io/github-action](https://github.com/cypress-io/github-action/). ( not sure how to do it )

thank you

**Comment by MikeMcC399** [comment]
@matart15 

> I will try running my test on CI without [cypress-io/github-action](https://github.com/cypress-io/github-action/). ( not sure how to do it )

I'm not sure that you need to do that.

- I have logged the issue about the error message in https://github.com/cypress-io/cypress/issues/25357 with a simple reproducible example, so that the Cypress experts can have a look at it.

**Comment by matart15** [comment]
@MikeMcC399 

oh i see. i misunderstood your message. thank you for reproducable example

**Comment by MikeMcC399** [comment]
@matart15 
Issue https://github.com/cypress-io/cypress/issues/25357 about the graphics error message is now routed to the Cypress e2e team.

**Comment by MikeMcC399** [comment]
@kahboom 

Are you still experiencing the issue you reported? If not, can this issue be closed now?

**Comment by kahboom** [comment]
No longer having this issue, thanks @MikeMcC399

**Comment by MikeMcC399** [comment]
@kahboom 

> No longer having this issue, 

Excellent! Many thanks for your confirmation.

**Comment by maxie7** [comment]
Hi, I have a problem with finding a cypress.config.ts file. Just with `npx cypress open` it works **fine**.

The problem is when I try to do it through docker-compose. I tried to copy cypress.config.ts file to /cypress folder. I also tried to investigate Cypress docs, no success.

![image](https://github.com/cypress-io/github-action/assets/15796607/b3bf3134-4609-4176-b8ab-0e519364f065)


How to define a path for cypress.config.ts in docker-compose file?!

**Comment by MikeMcC399** [comment]
@maxie7 

It does not look like your problem is related to `github-action` of this repository, so this is not the right place to get help. Also you have tagged on to a closed issue which is in any case not advisable to get new help.

You may find that somebody in the Cypress technical community on Discord can help you solve your problem:

[![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord)
(click on button)

**Comment by maxie7** [comment]
Thanks @MikeMcC399

**Comment by Sumbulmalik** [comment]
npx cypress open --config-file theNewCypressConfigPath         solved by this
